src/eve/visitors.py

```diff
@@ -227,104 +218,26 @@ def del_op(container: MutableSet, idx: int) -> None:
                 # Inmutable sequence or set: create a new container instance with the new values
                 tmp_items = [self.visit(value, **kwargs) for value in node]
                 result = node.__class__(  # type: ignore
-                    [value for value in tmp_items if value is not NOTHING]  # type: ignore
+                    [value for value in tmp_items if value is not concepts.NOTHING]  # type: ignore
                 )
 
             elif isinstance(node, collections.abc.Mapping):
                 # Inmutable mapping: create a new mapping instance with the new values
                 tmp_items = {key: self.visit(value, **kwargs) for key, value in node.items()}
                 result = node.__class__(  # type: ignore
-                    {key: value for key, value in tmp_items.items() if value is not NOTHING}
+                    {
+                        key: value
+                        for key, value in tmp_items.items()
+                        if value is not concepts.NOTHING
+                    }
                 )
 
             # Finally, in case current node object is mutable, process selected items (if any)
             for key, value in items:
                 new_value = self.visit(value, **kwargs)
-                if new_value is NOTHING:
+                if new_value is concepts.NOTHING:
                     del_op(result, key)
                 elif new_value != value:
                     set_op(result, key, new_value)
 
         return result
-
-
-@enum.unique
-class PathItemKind(enum.Enum):
-    CLASS = 1
-    COLLECTION = 2
-
-
-@dataclasses.dataclass(frozen=True)
-class PathItem:
-    node: AnyTreeNode
-    kind: PathItemKind
-    member: Union[str, int]
-
-
-@dataclasses.dataclass(frozen=True)
-class PathInfo:
-    items: List[PathItem] = dataclasses.field(default_factory=list)
-
-    def append(self, item: PathItem) -> "PathInfo":
-        return PathInfo([*self.items, item])
-
-    @property
-    def root(self) -> Optional[PathItem]:
-        return self.items[-1] if self.items else None
-
-    @property
-    def is_root(self) -> bool:
-        return len(self.items) == 0
-
-    def __iter__(self) -> Iterable[PathItem]:
-        return iter(self.items)
-
-
-class PathNodeVisitor:
```

Comment:
Make sense to remove this. Would be nice to have some sort of path selector/query facilities in the future.

Reply:
Working on it